File: Python/page_object/02_movemos_intenciones/home_page.py
from search_page import SearchPage


class HomePage(object):

    def __init__(self, driver):
        self._driver = driver
        self._url = 'http://www.mercadolibre.com.ar/'
        self._title = 'MercadoLibre Argentina'

    def search_for(self, keyword):
        search_field = self._driver.find_element_by_id("query")
        search_field.send_keys(keyword)
        search_field.submit()

        return SearchPage(self._driver)

File: Java/PageObject/04_MasTests/HomePage.java
package 04_MasTests;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import 04_MasTests.SearchPage;

public class HomePage {
	
	WebDriver _driver;
	String _url = "http://www.mercadolibre.com.ar/";
	String _title = "MercadoLibre Argentina";
	
	public HomePage(WebDriver driver) {
		this._driver = driver;
	}
	
	public SearchPage searchFor(String keyword){
        WebElement search_field = this._driver.findElement(By.id("query"));
        search_field.sendKeys(keyword);
        search_field.submit();

        return new SearchPage(this._driver);
	}
}
